sequencers/modeshape-sequencer-audio/src/main/java/org/modeshape/sequencer/audio/AudioMetadata.java

```
@@ -0,0 +1,301 @@
+            } else if (mimeType.startsWith("audio/x-wav") || mimeType.startsWith("audio/wav")) {
+                format = FORMAT_WAV;
+                audioFile = new WavFileReader().read(fileCopy);
+            }
+            return checkSupportedAudio();
+        } catch (Exception e) {
+            //log
+        }
+        return false;
+    }
+
+    /**
+     * Parse tags common for all audio files.
+     */
+    private boolean checkSupportedAudio() {
+            try {
+                AudioHeader header = audioFile.getAudioHeader();
+                bitrate = header.getBitRateAsNumber();
+                sampleRate = header.getSampleRateAsNumber();
+                channels = header.getChannels();
+                if (header.getChannels().toLowerCase().contains("stereo")) {
+                    channels = "2";
+                }
+                length = header.getTrackLength();
+
+                // generic frames
+                Tag tag = audioFile.getTag();
+                artist = tag.getFirst(FieldKey.ARTIST);
+                album = tag.getFirst(FieldKey.ALBUM);
+                title = tag.getFirst(FieldKey.TITLE);
+                comment = tag.getFirst(FieldKey.COMMENT);
+                year = tag.getFirst(FieldKey.YEAR);
+                track = tag.getFirst(FieldKey.TRACK);
+                genre = tag.getFirst(FieldKey.GENRE);
+
+                artwork = new ArrayList<AudioMetadataArtwork>();
+                for (Artwork a : tag.getArtworkList()) {
+                    AudioMetadataArtwork ama = new AudioMetadataArtwork();
+                    ama.setMimeType(a.getMimeType());
+                    if (a.getPictureType() >= 0) {
+                        ama.setType(a.getPictureType());
+                    }
+                    ama.setData(a.getBinaryData());
+
+                    artwork.add(ama);
+                }
+
+                return true;
+            } catch (Exception e) {
+                e.printStackTrace();
```

this should be logged or bubbled up instead of using `printStackTrace`